test/integration/BCowPool.t.sol

```diff
@@ -2,87 +2,67 @@
 pragma solidity 0.8.25;
 
 import {GPv2TradeEncoder} from './GPv2TradeEncoder.sol';
+import {PoolSwapIntegrationTest} from './PoolSwap.t.sol';
 import {IERC20} from '@cowprotocol/interfaces/IERC20.sol';
 import {GPv2Interaction} from '@cowprotocol/libraries/GPv2Interaction.sol';
 import {GPv2Order} from '@cowprotocol/libraries/GPv2Order.sol';
 import {GPv2Trade} from '@cowprotocol/libraries/GPv2Trade.sol';
 import {GPv2Signing} from '@cowprotocol/mixins/GPv2Signing.sol';
 import {BCoWConst} from 'contracts/BCoWConst.sol';
 import {BCoWPool} from 'contracts/BCoWPool.sol';
-import {BMath} from 'contracts/BMath.sol';
-import {GasSnapshot} from 'forge-gas-snapshot/GasSnapshot.sol';
-import {Test, Vm} from 'forge-std/Test.sol';
 import {IBCoWPool} from 'interfaces/IBCoWPool.sol';
-import {IBPool} from 'interfaces/IBPool.sol';
 import {ISettlement} from 'interfaces/ISettlement.sol';
 
-contract BCowPoolIntegrationTest is Test, BCoWConst, BMath, GasSnapshot {
+contract BCowPoolIntegrationTest is PoolSwapIntegrationTest, BCoWConst {
   using GPv2Order for GPv2Order.Data;
 
-  IBCoWPool public pool;
-
-  IERC20 public dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
-  IERC20 public weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
-  ISettlement public settlement = ISettlement(0x9008D19f58AAbD9eD0D60971565AA8510560ab41);
   address public solver = address(0xa5559C2E1302c5Ce82582A6b1E4Aec562C2FbCf4);
 
-  address public controller = makeAddr('controller');
-  Vm.Wallet swapper = vm.createWallet('swapper');
+  ISettlement public settlement = ISettlement(0x9008D19f58AAbD9eD0D60971565AA8510560ab41);
 
   bytes32 public constant APP_DATA = bytes32('exampleIntegrationAppData');
 
-  uint256 public constant HUNDRED_UNITS = 100 ether;
-  uint256 public constant ONE_UNIT = 1 ether;
-
-  function setUp() public {
+  function setUp() public override {
     vm.createSelectFork('mainnet', 20_012_063);
 
-    // deal controller
-    deal(address(dai), controller, HUNDRED_UNITS);
-    deal(address(weth), controller, HUNDRED_UNITS);
-
-    // deal swapper
-    deal(address(weth), swapper.addr, HUNDRED_UNITS);
-
-    vm.startPrank(controller);
-    // deploy
-    // TODO: deploy with BCoWFactory
-    pool = new BCoWPool(address(settlement));
-    // bind
-    dai.approve(address(pool), type(uint256).max);
-    weth.approve(address(pool), type(uint256).max);
-    IBPool(pool).bind(address(dai), HUNDRED_UNITS, 2e18);
-    IBPool(pool).bind(address(weth), HUNDRED_UNITS, 8e18);
+    deal(address(dai), lp, HUNDRED_UNITS);
+    deal(address(weth), lp, HUNDRED_UNITS);
+
+    deal(address(dai), swapper.addr, ONE_UNIT);
+
+    vm.startPrank(lp);
+    // TODO: deploy using BCoWFactory
+    pool = address(new BCoWPool(address(settlement)));
```

Comment:
check that @0xteddybear already provided us a BCoWFactory in #94  💘

Reply:
Nicee